src/RequestData.php

```diff
@@ -78,4 +90,34 @@ public function __toString()
 
         return $data;
     }
+
+    /**
+     * Processes a redirect request, updating internal values to represent the new request data.
+     *
+     * @param integer $code     HTTP status code for the redirect.
+     * @param string  $location The Location header received.
+     */
+    public function redirect($code, $location)
+    {
+        switch ($code) {
+
+            //These cases require that we switch to the GET method.
+            //@see https://github.com/bagder/curl/blob/cc28bc472ec421cec2ba26d653e53892998a248d/lib/transfer.c#L1736
+            case 301:
+            case 302:
+                $this->method = 'GET';
+
+            //Note: 303, 307, 308 status is not supported in HTTP/1.0.
+            // case 303:
+            // case 307:
+            // case 308:
+
+                //Of course switch the location.
+                $this->url = $location;
```

Comment:
Another way to do that is to use PSR7 and resolve two URI's with https://github.com/guzzle/psr7/blob/master/src/UriResolver.php